static/app/views/explore/metrics/metricsQueryParamsProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import type {ReactNode} from 'react';
+
+import {createDefinedContext} from 'sentry/utils/performance/contexts/utils';
+import {
+  MetricsFrozenContextProvider,
+  type MetricsFrozenContextProviderProps,
+} from 'sentry/views/explore/metrics/metricsFrozenContext';
+import {MultiMetricsQueryParamsProvider} from 'sentry/views/explore/metrics/multiMetricsQueryParams';
+
+const [
+  _MetricsAnalyticsPageSourceProvider,
+  _useMetricsAnalyticsPageSource,
+  MetricsAnalyticsPageSourceContext,
+] = createDefinedContext<string>({
+  name: 'MetricsAnalyticsPageSourceContext',
+});
+
+export const useMetricsAnalyticsPageSource = _useMetricsAnalyticsPageSource;
+
+interface MetricsQueryParamsProviderProps {
+  children: ReactNode;
+  freeze?: MetricsFrozenContextProviderProps;
+}
+
+export function MetricsQueryParamsProvider({
+  children,
+  freeze,
+}: MetricsQueryParamsProviderProps) {
+  return (
+    <MetricsAnalyticsPageSourceContext value="metrics">
+      <MetricsFrozenContextProvider {...freeze}>
+        <MultiMetricsQueryParamsProvider>{children}</MultiMetricsQueryParamsProvider>
+      </MetricsFrozenContextProvider>
```

Comment:
This doesn't seem quite right. The frozen contexts are per metric right? the `MultiMetricsQueryParamsProvider` contains the data for all the metric on the page.

I think you want to inject it somewhere in [here](https://github.com/getsentry/sentry/blob/master/static/app/views/explore/metrics/metricsQueryParams.tsx#L73-L85) so it's per metric and can contains all the matching trace ids for that metric.